language-support/hs/bindings/src/DA/Ledger/Services/MeteringReportService.hs

```diff
@@ -29,6 +30,22 @@ import GHC.Word (Word32)
 import Data.Time.Calendar (Day(..))
 import Data.Time.Clock (secondsToDiffTime, UTCTime(..))
 
+data MeteringRequestByDay = MeteringRequestByDay {
+  from :: Day
+, to :: Maybe Day
+, application :: Maybe ApplicationId
+} deriving (Show, Eq)
+
+instance ToJSON MeteringRequestByDay where
```

Comment:
I think you can kill the `ToJSON` instance of `MeteringRequest` in that case.

Reply:
actually do we need this type at all now?

Reply:
`MeteringRequest` is still used in the `MeteringReport` and is both received from the Http service _and_ used to serialize the output json for both underlying transports.

The `MeteringRequestByDay` is only sent to the Http service

Reply:
So we need to keep both the `ToJSON` instance and both request variants.